lua/toggleterm.lua

```diff
@@ -17,13 +17,32 @@ local preferences = {
   size = 12,
   shade_filetypes = {},
   shade_terminals = true,
+  persist_size = true,
   direction = "horizontal"
 }
 
 -----------------------------------------------------------
 -- State
 -----------------------------------------------------------
 local terminals = {}
+local persistent = {}
+
+--- Get the size of the split. Order of priority is as follows:
+---   1. The size argument is a valid number > 0
+---   2. There is persistent width/height information from prev open state
+---   3. Default/base case perference size
+---
+--- If `preferences.persist_size = false` then option `2` in the
+--- list is skipped.
+--- @param size number
+local function get_size(size)
+  if not preferences.persist_size then
+    return (size and size > 0) or preferences.size
```

Comment:
I think this will return the boolean if size is not nil rather than `size`
think it should look like the conditional below i.e.
```suggestion
    return (size and size > 0) and size or preferences.size
```

alternately to get past all the size checking you can save it to a variable and reuse it in both places i.e.
```lua
local valid_size = size ~= nil and size > 0

return valid_size and size or preferences.size
```

Reply:
I like your solution better thanks will implement that